internal/evaluator/rego_test.go

```diff
@@ -106,9 +106,97 @@ func TestOCIBlob(t *testing.T) {
 	}
 }
 
+func TestPURLIsValid(t *testing.T) {
+	cases := []struct {
+		name string
+		uri  *ast.Term
+		err  bool
+	}{
+		{
+			name: "success",
+			uri:  ast.StringTerm("pkg:rpm/fedora/curl@7.50.3-1.fc25?arch=i386&distro=fedora-25"),
+		},
+		{
+			name: "unexpected uri type",
+			uri:  ast.IntNumberTerm(42),
+			err:  true,
+		},
+		{
+			name: "malformed PURL string",
+			uri:  ast.StringTerm("pkg::rpm//fedora/curl7.50.3-1.fc25?arch=i386&distro=fedora-"),
+			err:  true,
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			ctx := context.Background()
+			bctx := rego.BuiltinContext{Context: ctx}
+
+			isValid, err := purlIsValid(bctx, c.uri)
+			require.NoError(t, err)
+			if c.err {
```

Comment:
Consider changing the `cases` struct to contain specify an `expected` attribute, then change this to:

```go
require.NotNil(t, isValid)
require.Equal(t, isValid, ast.BooleanTerm(c.expected))
```

That should work for both cases.

Reply:
That's definitely cleaner. Done.